src/SDL2.cs

```diff
@@ -43,15 +44,37 @@ public static class SDL
 
 		#region UTF8 Marshaling
 
-		internal static byte[] UTF8_ToNative(string s)
+		/* Used for stack allocated string marshaling. */
+		internal static int Utf8Size(string str)
 		{
-			if (s == null)
+			return str != null ? (str.Length * 4) + 1 : 0;
+		}
+		internal static unsafe byte* Utf8Encode(string str, byte* buffer, int bufferSize)
+		{
+			if (str == null)
```

Comment:
If I knew for sure which apis have nullable strings, we could pick and choose. I'm only using one of the APIs that allows for it, GetPrefPath, where org is optional.  There are others I'm less sure about like the MessageBox APIs where maybe title, and other text should be optional.

I'm not sure what paths in this API are hot, but those could use the Utf8Size/Utf8Encode methods that don't allow nulls to be passed in.  I'm happy to make those changes if you can guide my hand to avoid any regressions.

Reply:
We may have to take a broad approach to this... each case will probably end up like this:

```
byte* strPtr;
if (str == null)
{
    strPtr = IntPtr.Zero;
}
else
{
    strPtr = stackalloc byte[Utf8Len(str)];
    strPtr = Utf8Encode(strPtr);
}
Func(strPtr);
```

```
IntPtr strPtr;
if (str == null)
{
    strPtr = IntPtr.Zero;
}
else
{
    strPtr = Marshal.AllocHGlobal(Utf8Len(str));
    strPtr = (IntPtr) Utf8Encode((byte*) strPtr);
}
Func(strPtr);
if (strPtr != IntPtr.Zero)
{
    Marshal.FreeHGlobal(strPtr);
}
```

This is where I really wish we had macros in C#...

Reply:
I uploaded what I had in mind.  Most API functions look like they couldn't do what they are designed to do if you pass a null string, so most can be simpler code paths.